tests/functional/aws-node-sdk/lib/utility/website-util.js

```diff
@@ -1,6 +1,206 @@
+        assert(response.includes(elem),
+            `Expected but did not find '${elem}' in html`);
+    }
+}
+
+function _assertContainsHtml(responseBody) {
+    assert(responseBody.startsWith('<html>') &&
+        responseBody.includes('</html>'), 'Did not find html tags');
+}
+
+function _assertResponseHtml404(method, response, type, bucketName) {
+    assert.strictEqual(response.statusCode, 404);
+    if (method === 'HEAD') {
+        if (type !== '404-no-such-bucket'
+        && type !== '404-no-such-website-configuration'
+        && type !== '404-not-found') {
+            throw new Error('This 404 error is not checked in ' +
+            'checkHTML()');
+        }
+        // don't need to check HTML for head requests
+        return;
+    }
+    _assertContainsHtml(response.body);
+    _assertResponseHtml(response.body, 'title', '404 Not Found');
+    _assertResponseHtml(response.body, 'h1', '404 Not Found');
+    if (type === '404-no-such-bucket') {
+        _assertResponseHtml(response.body, 'ul', [
+            'Code: NoSuchBucket',
+            'Message: The specified bucket does not exist.',
+            `BucketName: ${bucketName}`,
+        ]);
+    } else if (type === '404-no-such-website-configuration') {
+        _assertResponseHtml(response.body, 'ul', [
+            'Code: NoSuchWebsiteConfiguration',
+            'Message: The specified bucket does not have a ' +
+            'website configuration',
+            `BucketName: ${bucketName}`,
+        ]);
+    } else if (type === '404-not-found') {
+        _assertResponseHtml(response.body, 'ul', [
+            'Code: NoSuchKey',
+            'Message: The specified key does not exist.',
+        ]);
+    } else {
+        throw new Error('This 404 error is not checked in ' +
+        'checkHTML()');
+    }
+}
+
+function _assertResponseHtml403(method, response, type) {
+    assert.strictEqual(response.statusCode, 403);
+    if (method === 'HEAD') {
+        if (type === '403-access-denied') {
+            assert.strictEqual(response.headers['x-amz-error-code'],
+            'AccessDenied');
+            assert.strictEqual(response.headers['x-amz-error-message'],
+            'Access Denied');
+        } else if (type !== '403-retrieve-error-document') {
+            throw new Error('This 403 error is not checked in ' +
+            'checkHTML()');
+        }
+    } else {
+        _assertContainsHtml(response.body);
+        _assertResponseHtml(response.body, 'title', '403 Forbidden');
+        _assertResponseHtml(response.body, 'h1', '403 Forbidden');
+        _assertResponseHtml(response.body, 'ul', [
+            'Code: AccessDenied',
+            'Message: Access Denied',
+        ]);
+        if (type === '403-retrieve-error-document') {
+            _assertResponseHtml(response.body, 'h3',
+            'An Error Occurred While Attempting to ' +
+            'Retrieve a Custom Error Document');
+            const startIndex = response.body.indexOf('</h3>') + 5;
```

Comment:
basically just asserting there is a `ul` element after the end of the `h3` element (for this error there are two ul elements, one before the `h3` and one after)
the plus 5 doesn't make much difference, it just means create the slice after the `</h3>` -- creating the slice starting from the `indexOf('</h3>')` would create the slice including `</h3>`